services/s3/src/test/java/software/amazon/awssdk/services/s3/S3EndpointResolutionTest.java

```diff
@@ -278,6 +344,15 @@ public void accelerateEnabled_UsesVirtualAddressingWithAccelerateEndpoint() thro
                               String.format("https://%s.s3-accelerate.amazonaws.com", BUCKET));
     }
 
+    /**
+     * S3 accelerate and FIPS should not be allowed in use together.
+     */
+    @Test(expected = IllegalStateException.class)
+    public void accelerateAndFips_Fails() throws Exception {
+        mockHttpClient.stubNextResponse(mockListObjectsResponse());
+        clientBuilder().fipsEnabled(true).serviceConfiguration(withAccelerateEnabled()).build().listObjects(r -> r.bucket(BUCKET));
```

Comment:
Can we verify the exception message to make sure the correct exception is thrown?

Reply:
Done